src/d/actor/d_a_tpota.cpp

```diff
@@ -5,71 +5,171 @@
+            emitters[count] = 0;
+        }
+        count++;
+    }
+    return 1;
 }
 
 /* 00000298-000002FC       .text make_ripple__9daTpota_cF4cXyz */
-void daTpota_c::make_ripple(cXyz) {
-    /* Nonmatching */
+void daTpota_c::make_ripple(cXyz i_position) {
+    dComIfGp_particle_set(dPa_name::ID_SCENE_82B0, &i_position,
+        NULL, NULL, 0xff, NULL, -1, NULL, NULL, NULL);
+    return;
 }
 
+
+
 /* 000002FC-00000354       .text check_water_h__9daTpota_cFP15JPABaseParticlef */
-void daTpota_c::check_water_h(JPABaseParticle*, float) {
-    /* Nonmatching */
+int daTpota_c::check_water_h(JPABaseParticle* i_ptcl, float i_position_y) {
+    
+    unknown_struct* unknown_struct = field_0x2C4;
+    int ret = 0;
+    if (i_position_y <= -230.0f) {
+        for (int i = 0; i < ARRAY_SIZE(field_0x2C4); i++, unknown_struct++) {
+            if (unknown_struct->ptcl == i_ptcl) {
+                if (unknown_struct->pos_y > -230.0f) {
+                    ret = 1;
+                }
+                break;
+            }
+        }
+    }
+    return ret;
 }
 
 /* 00000354-00000380       .text clear_splash__9daTpota_cFv */
 void daTpota_c::clear_splash() {
-    /* Nonmatching */
+    unknown_struct* unknown_struct = field_0x2C4;
+    int count = 30;
+
+    while (count != 0) {
+        unknown_struct->ptcl = NULL;
+        unknown_struct->pos_y = 0.0;
+        unknown_struct++;
+        count--;
+    }
+    return;
 }
 
 /* 00000380-000003F4       .text renew_splash__9daTpota_cFv */
 void daTpota_c::renew_splash() {
-    /* Nonmatching */
+    
+
+    if (emitters[1] != NULL) {
+        JSUPtrList* list = &emitters[1]->mActiveParticles;
+        unknown_struct *unknown_struct = field_0x2C4;
+        if(list != NULL){
+            clear_splash();
+            for (JSUPtrLink* link = list->getFirstLink(); link != NULL && (link != NULL); link = link->getNext()){
+                JPABaseParticle* particle = (JPABaseParticle*)link->getObjectPtr();
+                JGeometry::TVec3<f32> position;
+                particle->getGlobalPosition(position);
+                unknown_struct->ptcl = particle;
+                unknown_struct->pos_y = position.y;
+                unknown_struct++;
+            }   
+            
+        }
+
+    }
+    return;
 }
 
 /* 000003F4-000004C8       .text _execute__9daTpota_cFv */
 bool daTpota_c::_execute() {
-    /* Nonmatching */
+    float local_38;
+    if (emitters[1] != NULL){
+        JSUPtrList* list = &emitters[1]->mActiveParticles;
+        if(list != NULL){
+            for (JSUPtrLink* link = list->getFirstLink(); link != NULL && (link != NULL);
+            link = link->getNext()){
+                JPABaseParticle* particle = (JPABaseParticle *)link->getObjectPtr();
+                JGeometry::TVec3<f32> position;
+                particle->getGlobalPosition(position);
+                int idk = check_water_h(particle, position.y);
+                if(idk != 0){
```

Comment:
```suggestion
                if(check_water_h(particle, position.y)){
```
This is probably just a boolean return value, I assume this will still match